audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -101,299 +74,160 @@
         webEnvironment = WebEnvironment.RANDOM_PORT,
         properties = {"akka.discovery.config.services.kafka.from-config=false"}
 )
-@Import(KafkaContractTest.KafkaInitializer.Config.class)
-@ContextConfiguration(initializers = {KafkaContractTest.KafkaInitializer.class})
+@Import(KafkaInitializer.Config.class)
+@ContextConfiguration(initializers = {KafkaInitializer.class})
 @ActiveProfiles({"akka-test"})
 class KafkaContractTest {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaContractTest.class);
-    private static final ObjectMapper MAPPER = new ObjectMapper();
-
-    // Serialiser for upstream test input
-    static class RequestSerializer implements Serializer<JsonNode> {
-        @Override
-        public byte[] serialize(final String s, final JsonNode auditRequest) {
-            try {
-                return MAPPER.writeValueAsBytes(auditRequest);
-            } catch (JsonProcessingException e) {
-                throw new SerializationFailedException("Failed to serialize " + auditRequest.toString(), e);
-            }
-        }
-    }
 
     @Autowired
-    private TestRestTemplate restTemplate;
+    Serializer<String> keySerializer;
+    @Autowired
+    Serializer<JsonNode> valueSerializer;
     @Autowired
-    private ActorSystem akkaActorSystem;
+    ActorSystem akkaActorSystem;
     @Autowired
-    private Materializer akkaMaterializer;
+    Materializer akkaMaterializer;
     @Autowired
-    private ConsumerTopicConfiguration consumerTopicConfiguration;
+    KafkaContainer kafka;
     @Autowired
-    private AuditServiceConfigProperties auditServiceConfigProperties;
+    AuditServiceConfigProperties auditServiceConfigProperties;
+
     @SpyBean
     private AuditService auditService;
-    private int errorBefore;
-    private int successBefore;
-    private int after;
-
-    @BeforeEach
-    void setup() {
-        errorBefore = Arrays.stream(new File(".").listFiles())
-                .filter(file -> file.getName().startsWith("Error"))
-                .collect(Collectors.toSet()).size();
-        successBefore = Arrays.stream(new File(".").listFiles())
-                .filter(file -> file.getName().startsWith("Success"))
-                .collect(Collectors.toSet()).size();
-    }
 
-    @AfterAll
-    static void tearDown() {
-        Arrays.stream(new File(".").listFiles())
-                .filter(file -> (file.getName().startsWith("Success") || file.getName().startsWith("Error")))
-                .peek(file -> LOGGER.info("Deleting file {}", file.getName()))
-                .forEach(File::deleteOnExit);
+    private final Function<String, Integer> fileCount = (final String prefix) -> Arrays
```

Comment:
I think that the fileCount,  currentErrorCount and currentSuccessCount need to be initialised inside a BeforeEach method.  The counterpart for removing the files is in a AfterEach so it will be possible for these two processes to get out of sync.  Looks like there was an approach that did pair-up the initial counting of the files and the tear down on a per-test basis.  Why have we changed this?

Reply:
I've created a "BeforeEach" where these are now initialised.